### ISSUE TITLE
Project not saved after creation

### ISSUE THREAD

**Issue by JanC** [OpenRefine/OpenRefine#4952]
Hi,
I noticed a behaviour where a freshly created project is not saved.

### To Reproduce
- Start OpenRefine
- Create a new project usign e.g. a [sample csv file addresses.csv](https://people.sc.fsu.edu/~jburkardt/data/csv/addresses.csv)
- Optionally modify the first entry of the data
- Stop OpenRefine by closing the terminal window
- Start OpenRefine again
- Select Open Project ..

### Current Results
The  previously created OR project is not present in the list effectively loosing all changes made to the project.
The AppData folder contains a `1864769305521.project.corrupted` 

### Expected Behavior
The project is present 

### Screenshots

![openrefine](https://user-images.githubusercontent.com/707703/173175852-81098e90-7903-4dad-b469-48a26c254f84.gif)

### Versions
 - Operating System: Windows 11
 - Browser Version:  Chrome Version 102.0.5005.63 (Official Build) (64-bit) 
 - JRE or JDK Version: 
 - OpenRefine: 3.5.2

### Datasets
 [addresses.csv](https://people.sc.fsu.edu/~jburkardt/data/csv/addresses.csv)

### Additional context
I tried on macOS where it seems to be working fine.  I believe that closing the OpenRefine  window kills the app and there is no time to save the project. Should not OpenRefine save the project directly after it's created?

**Comment by JanC** [comment]
After debugging a bit, I noticed that the project's metadata are correctly created in memory of the `ProjectManager` class but they are not saved to file. 

I wonder if a fix could be to simply save the project just after importing it here

https://github.com/OpenRefine/OpenRefine/blob/e3efd4ec270bfa07a2575efa6ece11b6e269d105/main/src/com/google/refine/importing/ImportingUtilities.java#L1047-L1051

calling the `ensureProjectSaved` seems to fix the issue: 
```java
ProjectManager.singleton.registerProject(project, pm);
ProjectManager.singleton.ensureProjectSaved(project.id);
```

**Comment by tfmorris** [comment]
OpenRefine uses a timer-based persistence strategy. Saving the project immediately after it is created with save the initial data and metadata, but it will still lose the edits from "Optionally modify the first entry of the data" above. Killing the server process without allowing it to gracefully exit will always risk data loss, so should be avoided. 

It's been quite a while since I've used OpenRefine on Windows, but I suspect a ctrl/C in the terminal window would signal the OpenRefine server to shut down  gracefully, avoiding data loss.

**Comment by JanC** [comment]
>  I suspect a ctrl/C in the terminal window would signal the OpenRefine server to shut down gracefully, avoiding data loss.

probably yes. The problem is that most of the non savvy users will just close the Windows terminal window which seems to kill the process instantly

**Comment by thadguidry** [comment]
This will be solved once we finish #3221  (and if you have been pining to learn the Rust programming language, it's a great issue to work on!)
@JanC Shall we close this issue in favor of that overall solution - a native and better interface for running OpenRefine?

**Comment by JanC** [comment]
yes let's close it then 👍
